src/graphql-schema.js

```diff
@@ -95,32 +96,33 @@ export const resolvers = {
       // if (user && user.id) {
       //   throw new Error('Already logged in.')
       // }
-
       const session = driver.session()
-      const res = await session.run('MATCH (u:User {email: "' + email + '"}) RETURN u.id, u.slug, u.name, u.avatar, u.email, u.password, u.role LIMIT 1')
-      let u = res.records[0]._fields ? zipObject([
-        'id',
-        'slug',
-        'name',
-        'avatar',
-        'email',
-        'password',
-        'role'
-      ], res.records[0]._fields) : null
-      if (u) {
-        if (await bcrypt.compareSync(password, u.password)) {
-          delete u.password
-          u.avatar = fixUrl(u.avatar)
-          return Object.assign(u, {
-            token: generateJwt(u)
-          })
+      try {
+        const res = await session.run('MATCH (u:User {email: "' + email + '"}) RETURN u.id, u.slug, u.name, u.avatar, u.email, u.password, u.role LIMIT 1')
+        const u = res.records[0]._fields ? zipObject([
+          'id',
+          'slug',
+          'name',
+          'avatar',
+          'email',
+          'password',
+          'role'
+        ], res.records[0]._fields) : null
```

Comment:
I would check if `res.records.length <= 0` rather than `try + catch`.

Reply:
the try and catch also captures the situation where the user exists but password is not validated. we could switch from try and catch to test if res.records.length <=0 but in that case we would need also to cover the case where the password is wrong. another reason I see is that we are using res.records[0]._fields to retrieve the data from the DB but in other cases we are using record.get.  the try and catch provides a simple and unique way to catch errors while the other method will be specific to every resolver. what is wrong with try and catch ?